### PR TITLE
feat(cli): add scaffolding and code generators

### PR DESCRIPTION
**Pull request by aryeko**
## Type

- [x] `feat` — New feature
- [x] `fix` — Bug fix
- [ ] `refactor` — Code restructure (no behavior change)
- [x] `docs` — Documentation only
- [x] `test` — Test coverage
- [x] `chore` — Build, CI, tooling
- [ ] `perf` — Performance improvement

## Summary

This PR introduces an initial `modkit` CLI with scaffolding and code-generation commands. It adds `new app`, `new module`, `new provider`, and `new controller` flows, embedded templates, naming safety utilities, and AST utilities for module registration automation. It also includes design specs for CLI architecture and release/versioning SDLC.

## Changes

- Add CLI entrypoint and command tree under `cmd/modkit` and `internal/cli/cmd`.
- Add `modkit new app` scaffolding for a runnable starter service.
- Add `modkit new module`, `modkit new provider`, and `modkit new controller` generators.
- Add embedded templates under `internal/cli/templates`.
- Add AST helper in `internal/cli/ast/modify.go` with explicit insertion failure handling and safe temp-file write+replace.
- Add naming sanitization utilities for package/type identifiers.
- Enforce provider token format in AST insertion (`module.component`).
- Fix generated template behavior and docs to match actual CLI output.
- Add extensive CLI and AST tests for success/error branches and edge cases.
- Add specs:
  - `docs/specs/design-cli-tooling.md`
  - `docs/specs/design-release-versioning-sdlc-cli.md`
- Add `dist/` to `.gitignore`.

## Breaking Changes

None.

## Validation

```bash
make fmt
make lint
make vuln
make test
make test-coverage
make test-patch-coverage
```

All commands passed locally.
Patch coverage now reports ~89% locally (`240/269`) and Codecov patch check is passing.

## Checklist

- [x] Code follows project style (`make fmt` passes)
- [x] Linter passes (`make lint`)
- [x] Tests pass (`make test`)
- [x] Tests added/updated for new functionality
- [x] Documentation updated (if applicable)
- [x] Commit messages follow [Conventional Commits](https://www.conventionalcommits.org/)

## Resolves

N/A

## Notes

- Provider/controller commands currently generate files and print explicit TODO registration guidance for `module.go`.
- `dist/` build artifacts are intentionally untracked and excluded from this PR.

## Release Notes

* **New Features**
  * Introduced new CLI scaffolding tool with commands to generate applications, modules, providers, and controllers
  * Includes code templates and automatic project initialization

* **Documentation**
  * Added comprehensive design specifications for CLI tooling and release versioning

### CODE REVIEW

File: internal/cli/cmd/new_provider.go [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+		return fmt.Errorf("failed to close file: %w", closeErr)
+	}
+
+	fmt.Printf("Created %s\n", providerPath)
+
+	tokenName := fmt.Sprintf("%s.service", strings.ToLower(name))
+
+	fmt.Printf("TODO: Register provider in %s:\n", modulePath)
+	fmt.Printf("  Token: %q\n", tokenName)
+	fmt.Printf("  Build: func(r module.Resolver) (any, error) { return New%sService(), nil }\n", data.Title(name))
```
[the remark this answers is not in the document]
Comment (aryeko) [reply internal/cli/cmd/new_provider.go]: Resolved in this PR: provider token now follows `module.component` convention using module package and provider name.

Implemented at `internal/cli/cmd/new_provider.go`:

```go
tokenName := fmt.Sprintf("%s.%s", pkgName, strings.ToLower(name))
```

This yields tokens such as `users.cache`.
